src/appleseed/foundation/platform/system.cpp

```diff
@@ -421,6 +429,388 @@ uint64 System::get_process_virtual_memory_size()
     return static_cast<uint64>(rss) * sysconf(_SC_PAGESIZE);
 }
 
+// ------------------------------------------------------------------------------------------------
+// FreeBSD.
+// ------------------------------------------------------------------------------------------------
```

Comment:
Very nice code!

Regarding the formatting and coding style, for the sake of consistency there are a few minor things that could be improved:

* Wrap utility types and functions into an anonymous namespace, instead of making them `static`.
* Classes/structures are spelled in CamelCase.
* Macros are spelled in UPPERCASE.
* The pointer `*` goes on the type's side rather than the variable's side.
* We don't repeat the `struct` keyword, which is optional in C++.